st2common/st2common/models/system/action.py

```diff
@@ -210,6 +186,36 @@ def _get_script_arguments(self, named_args=None, positional_args=None):
             command_parts.append(positional_args)
         return ' '.join(command_parts)
 
+    @staticmethod
+    def _generate_command_string(script_path_abs, script_arguments, is_sudo=False,
+                                 user=LOGGED_USER_USERNAME):
+        if is_sudo:
+            if script_arguments:
+                command = quote_unix('%s %s' % (script_path_abs, script_arguments))
+            else:
+                command = quote_unix(script_path_abs)
+
+            command = 'sudo -E -- bash -c %s' % (command)
+        else:
+            if user and user != LOGGED_USER_USERNAME:
+                # Need to use sudo to run as a different user
+                user = quote_unix(user)
+
+                if script_arguments:
+                    command = quote_unix('%s %s' % (script_path_abs, script_arguments))
+                else:
+                    command = quote_unix(script_path_abs)
+
+                command = 'sudo -E -u %s -- bash -c %s' % (user, command)
```

Comment:
This is actually confusing for me. (I just moved the code). Since now we can now connect to boxes as particular user and private key/password, I wonder if this `sudo` is needed here? Unfortunately, this might be needed in Fabric case but in paramiko case, I am thinking this sudo as user isn't needed. Thoughts?